pyaedt/modeler/cad/Primitives.py

```diff
@@ -275,45 +587,4908 @@ def version(self):
+        )
+
+        objs = self._all_object_names
+        self.oeditor.SweepFacesAlongNormal(vArg1, vArg2)
+        self.cleanup_objects()
+        objs2 = self._all_object_names
+        obj = [i for i in objs2 if i not in objs]
+        for el in obj:
+            self._create_object(el)
+        if obj:
+            if len(obj) > 1:
+                return [self.update_object(self[o]) for o in obj]
+            else:
+                return self.update_object(self[obj[0]])
+        return False
+
+    @pyaedt_function_handler()
+    def sweep_along_vector(self, objid, sweep_vector, draft_angle=0, draft_type="Round"):
+        """Sweep the selection along a vector.
+
+        Parameters
+        ----------
+        objid : list, str, int, :class:`pyaedt.modeler.Object3d.Object3d`
+            Name or ID of the object.
+        sweep_vector : float
+            List of ``[x1, y1, z1]`` coordinates or Application.Position object for
+            the vector.
+        draft_angle : float, optional
+            Draft angle in degrees. The default is ``0``.
+        draft_type : str
+            Type of the draft. Options are ``"Round"``, ``"Natural"``,
+            and ``"Extended"``. The default is ``"Round"``.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+
+        >>> oEditor.SweepAlongVector
+        """
+        selections = self.convert_to_selections(objid)
+        vectorx, vectory, vectorz = self._pos_with_arg(sweep_vector)
+        vArg1 = ["NAME:Selections", "Selections:=", selections, "NewPartsModelFlag:=", "Model"]
+        vArg2 = ["NAME:VectorSweepParameters"]
+        vArg2.append("DraftAngle:="), vArg2.append(self._arg_with_dim(draft_angle, "deg"))
+        vArg2.append("DraftType:="), vArg2.append(GeometryOperators.draft_type_str(draft_type))
+        vArg2.append("SweepVectorX:="), vArg2.append(vectorx)
+        vArg2.append("SweepVectorY:="), vArg2.append(vectory)
+        vArg2.append("SweepVectorZ:="), vArg2.append(vectorz)
+
+        self.oeditor.SweepAlongVector(vArg1, vArg2)
+
+        return self.update_object(objid)
+
+    @pyaedt_function_handler()
+    def sweep_along_path(
+        self, objid, sweep_object, draft_angle=0, draft_type="Round", is_check_face_intersection=False, twist_angle=0
+    ):
+        """Sweep the selection along a path.
+
+        Parameters
+        ----------
+        objid : list, str, int, :class:`pyaedt.modeler.Object3d.Object3d`
+            Name or ID of the object.
+        sweep_object : str, int
+            Name or ID of the sweep.
+        draft_angle : float, optional
+            Draft angle in degrees. The default is ``0``.
+        draft_type : str
+            Type of the draft. Options are ``"Round"``, ``"Natural"``,
+            and ``"Extended"``. The default is ``"Round"``.
+        is_check_face_intersection : bool, optional
+            The default is ``False``.
```

Comment:
```suggestion
        draft_type : str
            Type of the draft. Options are ``"Extended"``, ``"Natural"``,
            and ``"Round"``. The default is ``"Round"``.
        is_check_face_intersection : bool, optional
            The default is ``False``.
```
A description must be added for ``is_check_face_intersection``.